llama_index/embeddings/huggingface.py

```diff
@@ -1,15 +1,26 @@
-from typing import Any, List, Optional
+import asyncio
+from typing import TYPE_CHECKING, Any, List, Optional
+
+from langchain.schema.embeddings import Embeddings
 
 from llama_index.bridge.pydantic import Field, PrivateAttr
 from llama_index.callbacks import CallbackManager
-from llama_index.embeddings.base import DEFAULT_EMBED_BATCH_SIZE, BaseEmbedding
+from llama_index.embeddings.base import (
+    DEFAULT_EMBED_BATCH_SIZE,
+    BaseEmbedding,
+    Embedding,
+)
 from llama_index.embeddings.huggingface_utils import (
     DEFAULT_HUGGINGFACE_EMBEDDING_MODEL,
     get_query_instruct_for_model_name,
     get_text_instruct_for_model_name,
 )
+from llama_index.llms.huggingface import HuggingFaceInferenceAPI
 from llama_index.utils import get_cache_dir
 
+if TYPE_CHECKING:
+    from llama_index.embeddings.utils import Pooling
+
 
 class HuggingFaceEmbedding(BaseEmbedding):
```

Comment:
Future PR -- lets use the Pooling class in HugggingFaceEmbeddings and Optimum embeddings

Reply:
Agreed! Can expand it to be input `torch` tensors as well